### DISCUSSION TITLE
getrawtransaction not returning tx info!

### DISCUSSION THREAD

**Discussion by KnsDev330** [dogecoin/dogecoin#3132] [Help needed] [answered]
Configuration file
```
daemon=1
server=1
listen=1
rpcuser=***
rpcpassword=***
prune=4096
datadir=/var/www/nodes/.dogecoin
```
on block 4428361 (`4277b53cffeb707c44eb7881e07cf03678a21f83b22bdbd8fbf49c3f6dc028ac`)
this transaction hash `d33a775d4c791fc192b918b2254b920b9cd93a40a9c212732125b1b382ab6573` is not returning any transaction info. 
It say's
```
error code: -5
error message:
No such mempool transaction. Use -txindex to enable blockchain transaction queries. Use gettransaction for wallet transactions.
```

but other hashes from the block returns tx info.
command I ran was ```dogecoin-cli getrawtransaction d33a775d4c791fc192b918b2254b920b9cd93a40a9c212732125b1b382ab6573 true```

It tells to use `-txindex` but as I am using pruned node I can't do that.
But then why it returns other tx info's?

Dogecoin Core 1.14.6
Debian 11

**Comment by patricklodder** [comment, marked as answer]
A node without `-txindex` only tracks the mempool, unspent outputs and, if it has a wallet, transactions belonging to the wallet. 

Per the error message, if this is a transaction that includes any of your wallet addresses, you can examine it with `gettransaction` instead. 

The reason why you can sometimes get information about a transaction that is not in the mempool or wallet is because one of its outputs is still unspent. This is old functionality that is marked for depreciation and you shouldn't rely on it (also because if an output gets spent, it no longer works). From `dogecoin-cli help getrawtransaction`:

> NOTE: By default this function only works for mempool transactions. If the -txindex option is
enabled, it also works for blockchain transactions.
> DEPRECATED: for now, it also works for transactions with unspent outputs.

The only way to be able to query every transaction is by enabling `-txindex` and yes, that means no pruning support. What is your use-case?
